kernel/src/fs/inode_handle/mod.rs

```diff
@@ -167,6 +167,21 @@ impl InodeHandle_ {
         self.path.inode().ioctl(cmd, arg)
     }
 
+    fn mappable(&self) -> Result<Mappable> {
+        let inode = self.path.inode();
+        if inode.page_cache().is_some() {
+            // If the inode has a page cache, it is a file-backed mapping and
+            // we directly return the corresponding inode.
+            Ok(Mappable::Inode(inode.clone()))
+        } else if let Some(ref file_io) = self.file_io {
+            // Otherwise, it is a special file (e.g. device file) and we should
+            // return the file-specific mappable object.
+            file_io.mappable()
+        } else {
+            return_errno_with_message!(Errno::EINVAL, "The file is not mappable");
```

Comment:
- Lowercase the first letter in the error message:
```suggestion
            return_errno_with_message!(Errno::EINVAL, "the file is not mappable");
```

- I think since you change this, you should change the error messages in the default implementation of `FileIo::mapped` and `FileLike::mapped` as well. Any reason not to do that?